code/game/machinery/autolathe.dm

```diff
@@ -43,17 +43,57 @@
 		GLOB.autounlock_techwebs[/datum/techweb/autounlocking/autolathe] = new /datum/techweb/autounlocking/autolathe
 	stored_research = GLOB.autounlock_techwebs[/datum/techweb/autounlocking/autolathe]
 
+	register_context()
+
 /obj/machinery/autolathe/Destroy()
 	materials = null
 	QDEL_NULL(wires)
 	return ..()
 
+/obj/machinery/autolathe/examine(mob/user)
+	. = ..()
+	if(in_range(user, src) || isobserver(user))
+		. += span_notice("The status display reads: Storing up to <b>[materials.max_amount]</b> material units.<br>Material consumption at <b>[creation_efficiency*100]%</b>.")
+		if(drop_direction)
+			. += span_notice("Currently configured to drop printed objects <b>[dir2text(drop_direction)]</b>.")
+			. += span_notice("[EXAMINE_HINT("Alt-click")] to reset.")
+		else
+			. += span_notice("[EXAMINE_HINT("Drag")] towards a direction (while next to it) to change drop direction.")
+
+		. += span_notice("Its maintainence panel can be [EXAMINE_HINT("screwed")] [panel_open ? "closed" : "open"].")
+		if(panel_open)
+			. += span_notice("The machine can be [EXAMINE_HINT("pried")] apart.")
+
+/obj/machinery/autolathe/add_context(atom/source, list/context, obj/item/held_item, mob/user)
+	if(isnull(held_item) && drop_direction)
```

Comment:
You sanity check for `isnull(held_item)`, but only when `drop_direction` is set. 

So this proc will runtime if `held_item` is `null`, but `drop_direction` is not set, later down the line.

Reply:
Also, ostensibly, you can alt-click on things while holding items. So this is isn't necessarily accurate.